src/lib.rs

```diff
@@ -20,16 +24,37 @@ pub struct Config {
     pub redirect_url: String,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd)]
+// See https://tools.ietf.org/html/rfc6749#section-5.1
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Deserialize)]
 pub struct Token {
     pub access_token: String,
+    #[serde(default)]
     pub scopes: Vec<String>,
     pub token_type: String,
+    #[serde(default)]
+    pub expires_in: Option<u32>,
+    #[serde(default)]
+    pub refresh_token: Option<String>,
+}
+
+// See https://tools.ietf.org/html/rfc6749#section-5.2
+#[derive(Debug, Deserialize)]
+struct TokenError {
```

Comment:
[crates.io is](https://github.com/rust-lang/crates.io/blob/9ec474cc62e77c35a88fe25a31cbb44d27a5cc72/src/http.rs#L70-L78) :(